src/groundlight/internalapi.py

```diff
@@ -67,9 +71,76 @@ def iq_is_confident(iq: ImageQuery, confidence_threshold: float) -> bool:
     return iq.result.confidence >= confidence_threshold
 
 
-class InternalApiError(RuntimeError):
-    # TODO: We need a better exception hierarchy
-    pass
+class InternalApiError(ApiException, RuntimeError):
+    # TODO: We should really avoid this double inheritance since
```

Comment:
I'd prefer to clean this up rather than make it uglier.  Changing the base class technically is a breaking change though.  So maybe we leave that for another PR.  This one is getting long in the tooth already.

Reply:
I can submit a follow up PR sometime this week cleaning this up.